db/db_impl.cc

```diff
@@ -645,6 +668,7 @@ Status DBImpl::SyncWAL() {
     need_log_dir_sync = !log_dir_synced_;
   }
 
+  TEST_SYNC_POINT("DBWALTest::SyncWALNotWaitWrite:1");
```

Comment:
I believe every access to logs_ should grab log_write_mutex_. In SyncWAL(), it is not the case.